### DISCUSSION TITLE
What about replace Hackney with Tesla?

### DISCUSSION THREAD

**Discussion by nallwhy** [mgallo/openai.ex#45] [Ideas]
### Describe the feature or improvement you're requesting

Tesla is easier to control than Hackney.
(ex. http2, retrying, ...)

What about replace Hackney with Tesla, and set default Tesla adapter as Hackney adapter?

**Comment by nallwhy** [comment]
It can be a breaking change, but I think it's worth it.

**Comment by mgallo** [comment]
hey, thanks for the suggestion. It would be really cool, especially the use of middleware would simplify part of the client and make it more flexible. 
I will evaluate this, but I certainly won't be able to do anything anytime soon. If so, I will try to see if it is possible to integrate it into a first major version (maybe with a compatibility layer to be retrocompatible with the old apis)

**Comment by michaelterryio** [comment]
I would rather Req: https://github.com/wojtekmach/req

**Comment by nallwhy** [comment]
These would be helpful.

https://elixirforum.com/t/mint-vs-finch-vs-gun-vs-tesla-vs-httpoison-etc/38588/11
https://elixirforum.com/t/mint-vs-finch-vs-gun-vs-tesla-vs-httpoison-etc/38588/14
